SourcerySwift/Sources/SwiftTemplate.swift

```diff
@@ -186,22 +187,32 @@ open class SwiftTemplate {
             }
             processedComponents.append(component)
         }
-
+        Log.benchmark("\tRaw command processing for \(sourcePath.lastComponent) took: \(currentTimestamp() - startProcessing)")
         return commands
     }
 
     public func render(_ context: Any) throws -> String {
         let binaryPath: Path
 
         if let cachePath = cachePath,
-            let hash = cacheKey,
-            let hashPath = hash.addingPercentEncoding(withAllowedCharacters: CharacterSet.alphanumerics) {
-
+           let hash = executableCacheKey,
+           let hashPath = hash.addingPercentEncoding(withAllowedCharacters: CharacterSet.alphanumerics) {
             binaryPath = cachePath + hashPath
-            if !binaryPath.exists {
+            if binaryPath.exists {
+                Log.verbose("Reusing built SwiftTemplate binary for SwiftTemplate with cache key: \(hash)...")
+            } else {
+                Log.verbose("Building new SwiftTemplate binary for SwiftTemplate...")
                 try? cachePath.delete() // clear old cache
                 try cachePath.mkdir()
-                try build().move(binaryPath)
+                do {
+                    try build().move(binaryPath)
+                } catch let error as NSError {
+                    if error.domain == "NSCocoaErrorDomain", error.code == 516 {
```

Comment:
This is what now allows to execute Sourcery from multiple processes in parallel, generating code into different directories when `swifttemplate` is shared across these executions (i.e. cached and reused).